packages/tx-construction/test/tx-builder/TxBuilderDelegatePortfolio.test.ts

```diff
@@ -292,9 +294,23 @@ describe('TxBuilder/delegatePortfolio', () => {
       groupedAddresses = txBuilderFactory.groupedAddresses;
       txBuilder = txBuilderFactory.txBuilder;
       txBuilderProviders = txBuilderFactory.txBuilderProviders;
+
+      nonDelegatingWalletTxBuilder = (
+        await createTxBuilder({
+          keyAgent,
+          stakeKeyDelegations: []
```

Comment:
Could you also include 2 unregistered stake keys to the scenario, so we test that it ignores them and uses random improve?